lib/tz_world.ex

```diff
@@ -1,89 +1,139 @@
 defmodule TzWorld do
   @moduledoc """
   Resolve a timezone name from coordinates.
+
   """
   use GenServer
-  alias :math, as: Math
   alias Geo.Point
+  import TzWorld.Guards
+  alias TzWorld.GeoData
 
-  @data_archive Application.app_dir(:tz_world, "priv/timezones-data.zip")
+  @timeout 10_000
 
   @doc false
-  def start_link do
+  def start_link(_) do
     GenServer.start_link(__MODULE__, [], name: __MODULE__)
   end
 
   def init(_state) do
-    Process.send_after(self(), :init_state, 0)
-    {:ok, []}
+    {:ok, [], {:continue, :load_data}}
+  end
+
+  def version do
+    GenServer.call(__MODULE__, :version, @timeout)
   end
 
   @doc """
-  Returns the timezone name for the given coordinates.
+  Returns the timezone name for the given coordinates specified
+  as either a `Geo.Point` or as `lng` and `lat` parameters
 
   ## Examples
 
-      iex(1)> TzWorld.timezone_at(%Geo.Point{coordinates: {3.2, 45.32}})
-      "Europe/Paris"
+      iex> TzWorld.timezone_at(%Geo.Point{coordinates: {3.2, 45.32}})
+      {:ok, "Europe/Paris"}
+
+      iex> TzWorld.timezone_at(3.2, 45.32)
+      {:ok, "Europe/Paris"}
+
+
+  The algorithm starts by filtering out timezones whose bounding
+  box does not contain the given point.
+
+  Once filtered, the first timezone which contains the given
+  point is returned, or `nil` if none of the timzones match.
 
-  The algorithm starts by filtering out timezones whose bounding box does not contain
-  the given point.
-  Once filtered, we return the first timezone which contains the given point, or `nil` if
-  none of the timzones match.
   """
-  @spec timezone_at(Geo.Point.t) :: String.t | nil
+  @spec timezone_at(Geo.Point.t()) :: String.t() | nil
```

Comment:
Is this spec still true? I believe it returns a tuple now, right (the comment also applies to the docblock)?

Reply:
Updated the spec.